checker.go

```diff
@@ -305,6 +305,51 @@ func (c *isNilChecker) Negate(got interface{}, args []interface{}) error {
 	return errors.New("the value is nil, but should not")
 }
 
+// HasLen is a Checker checking that the provided value has the provided length.
+// For instance:
+//
+//     c.Assert([]int{42, 47}, qt.HasLen, 2)
+//     c.Assert(myMap, qt.HasLen, 42)
+//
+var HasLen Checker = &hasLenChecker{
+	numArgs: 1,
+}
+
+type hasLenChecker struct {
+	numArgs
+}
+
+// Check implements Checker.Check by checking that len(got) == args[0].
+func (c *hasLenChecker) Check(got interface{}, args []interface{}) (err error) {
+	want, ok := args[0].(int)
+	if !ok {
+		return BadCheckf("expected a numeric length to compare the value to, got %T instead", args[0])
```

Comment:
"numeric length" isn't very specific.


How about: "expected length is of type %T, not int" ?

Reply:
Done.